server/index.js

```diff
@@ -58,37 +60,53 @@ const addAssets = async (app) => {
     return;
   }
   if (isDevelopment) {
-    await app.register(fastifyWebpackHMR, {
+    app.register(fastifyWebpackHMR, {
       config: path.join(__dirname, '..', 'webpack.config.js'),
     });
   } else {
-    await app.register(fastifyStatic, {
+    app.register(fastifyStatic, {
       root: path.join(__dirname, '..', 'assets'),
       prefix: '/assets',
     });
   }
 };
 const addDatabase = async (app) => {
-  await app.register(fastifyObjection, {
+  app.register(fastifyObjection, {
     knexConfig: knexConfig[mode],
     models,
   });
 };
 const addPlugins = async (app) => {
-  await app.register(fastifyFormbody);
-  await app.register(fastifyMethodOverride);
+  app.register(fastifyFormbody);
+  app.register(fastifyMethodOverride);
 };
-const addSession = async (app) => {
-  await app.register(fastifySecureSession, {
+const addAuthentification = async (app) => {
+  app.register(fastifySecureSession, {
     secret: process.env.SECRET,
     salt: process.env.SALT,
   });
-  app.addHook('preHandler', async (req) => {
-    const userId = req.session.get('userId');
-    if (userId !== undefined) {
-      req.currentUser = await app.objection.models.user.query().findById(userId);
-    }
-  });
+  app.register(fastifyPassport.initialize());
+  app.register(fastifyPassport.secureSession());
+  fastifyPassport.use('form', new FormPassportStrategy('form', app));
+  fastifyPassport.registerUserDeserializer(
+    async (user) => app.objection.models.user.query().findById(user.id),
+  );
+  fastifyPassport.registerUserSerializer(async (user) => user);
+  app.decorate('pasport', fastifyPassport);
+  app.decorate('formAuth', (...args) => fastifyPassport.authenticate(
```

Comment:
why form? it would work even if we'd send json to api, wouldn't it?

Reply:
because its using `form` strategy?

Reply:
okay